tests/unittests/benchmark/task/profet/test_profet_task.py

```diff
@@ -0,0 +1,375 @@
+
+    min_y, max_y = y_min[benchmark], y_max[benchmark]
+    assert min_y <= real_y.min() and min_y <= fake_y.min()
+    assert real_y.max() <= max_y and fake_y.max() <= max_y
+
+    min_c, max_c = c_min[benchmark], c_max[benchmark]
+    assert min_c <= real_c.min() and min_c <= fake_c.min()
+    assert real_c.max() <= max_c and fake_c.max() <= max_c
+
+
+class ProfetTaskTests:
+    """ Base class for testing Profet tasks. """
+
+    # TODO: What would be a good set of tests for the profet tasks?
+
+    Task: ClassVar[Type[ProfetTask]]
+
+    @pytest.mark.timeout(30)
+    def test_instantiating_task(
+        self,
+        profet_train_config: MetaModelTrainingConfig,
+        profet_input_dir: Path,
+        tmp_path_factory,
+    ):
+        """ TODO: Test that when instantiating multiple tasks with the same arguments, the
+        meta-model is trained only once.
+        """
+        max_trials = 123
+        task_id = 0
+        seed = 123
+
+        checkpoint_dir: Path = tmp_path_factory.mktemp("checkpoints")
+        # Directory should be empty.
+        assert len(list(checkpoint_dir.iterdir())) == 0
+
+        task = self.Task(
+            max_trials=max_trials,
+            task_id=task_id,
+            train_config=profet_train_config,
+            seed=seed,
+            input_dir=profet_input_dir,
+            checkpoint_dir=checkpoint_dir,
+        )
+        assert task.max_trials == max_trials
+        assert task.seed == seed
+        assert task.task_id == task_id
+
+        # Directory should have one file (the trained model).
+        assert len(list(checkpoint_dir.iterdir())) == 1
+
+        task = self.Task(
+            max_trials=max_trials,
+            task_id=task_id,
+            train_config=profet_train_config,
+            seed=seed,
+            input_dir=profet_input_dir,
+            checkpoint_dir=checkpoint_dir,
+        )
+        assert task.max_trials == max_trials
+        assert task.seed == seed
+        assert task.task_id == task_id
+
+        # Directory should have one file (the trained model).
+        assert len(list(checkpoint_dir.iterdir())) == 1
+
+    def test_configuration(
+        self,
+        profet_train_config: MetaModelTrainingConfig,
+        profet_input_dir: Path,
+        checkpoint_dir: Path,
+    ):
+        """ Test that tasks have a proper configuration and that they can be created from it.
+        """
+        task_id = 0
+        kwargs = dict(
+            max_trials=10,
+            task_id=task_id,
+            train_config=profet_train_config,
+            seed=123,
+            input_dir=profet_input_dir,
+            checkpoint_dir=checkpoint_dir,
+        )
+        first_task = self.Task(
+            **kwargs
+        )
+        configuration = first_task.configuration
+        assert len(configuration.keys()) == 1
+        key, config_dict = configuration.popitem()
+        assert key == self.Task.__qualname__
+        
+        for key, value in kwargs.items():
```

Comment:
Why not plain simple `assert config_dict == kwargs`?

Reply:
because the config might have more entries than the kwargs

Reply:
Are they tested? In this case I would pop the keys from config_dict that are tested and then do `assert config_dict == {}` to make sure we tested all items. If they are not part of the constructor, they should be tested after comparing with kwargs.

Reply:
If some tasks have special config to test, this can be done in a method called here that can be overwritten by child test suite.

Reply:
Sorry, I don't quite understand.. What do you mean by "are they tested"?
I'd appreciate some help on this, if you have the time/energy obviously.

Reply:
Oh sorry for my lack of clarity! What I meant is that, if the config has more entries than the kwargs, than when we compare a subset of config with kwargs, the additional parameters of the config are not tested. My question was, are these values tested somewhere else?

Reply:
Fixed: Taking a simpler approach. The config should match the kwargs, in string/serialized form.